sentiment_analysis.ipynb

```diff
@@ -0,0 +1,3780 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 98,
+   "id": "88PnAgVfpe0f",
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "88PnAgVfpe0f",
+    "outputId": "2eb02321-40bd-4631-af47-56b70ac17b2e"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(992, 6)"
+      ]
+     },
+     "execution_count": 98,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "output.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 134,
+   "id": "Knkwz89lo0E1",
+   "metadata": {
+    "id": "Knkwz89lo0E1"
+   },
+   "outputs": [],
+   "source": [
+    "sentiment_pred = []\n",
+    "\n",
+    "classify = lambda lyrics: max(classifier(lyrics)[0], key=lambda x: x['score'])['label']\n",
+    "def classify_with_error(lyrics, i):\n",
+    "  try:\n",
+    "    return classify(lyrics)\n",
+    "  except Exception as e:\n",
+    "    print(i,\":\",e)\n",
+    "    return None\n",
+    "\n",
+    "# for row in range(len(output)):\n",
+    "#   print(f\"row: {row}\")\n",
+    "#   sentiment_pred.append(classify_with_error(reordered_df.iloc[row][-1]))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 110,
+   "id": "Diz7zoHJrSic",
+   "metadata": {
+    "id": "Diz7zoHJrSic"
+   },
+   "outputs": [],
+   "source": [
+    "reordered_df['lengths'] = reordered_df['lyrics'].apply(len)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 140,
+   "id": "TAAgoHG4rzKk",
+   "metadata": {
+    "id": "TAAgoHG4rzKk"
+   },
+   "outputs": [],
+   "source": [
+    "max_length = 500  # start with 1000 and trim based on error\n",
+    "truncate = lambda x: x[:max_length] if len(x) > max_length else x\n",
+    "reordered_df['truncated_lyrics'] = reordered_df['lyrics'].apply(truncate)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 137,
+   "id": "f7mgMpgZzj0k",
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "f7mgMpgZzj0k",
+    "outputId": "f4360205-74c0-450c-c466-0596fdc3e219"
+   },
+   "outputs": [
+    {
+     "metadata": {
+      "tags": null
+     },
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "92 : The size of tensor a (523) must match the size of tensor b (512) at non-singleton dimension 1\n",
+      "125 : The size of tensor a (515) must match the size of tensor b (512) at non-singleton dimension 1\n",
+      "459 : The size of tensor a (611) must match the size of tensor b (512) at non-singleton dimension 1\n",
+      "577 : The size of tensor a (632) must match the size of tensor b (512) at non-singleton dimension 1\n"
+     ]
+    }
+   ],
+   "source": [
+    "sentiment_pred = [\"\"]*len(output)\n",
+    "for i in range(len(output)):\n",
+    "  sentiment_pred[i] = classify_with_error(reordered_df['truncated_lyrics'].iloc[i],i)"
```

Comment:
What is the purpose of this function?

Reply:
seems to be try-catching the library classify function? did you figure out which ones were causing errors?